### PR TITLE
MULE-11686: Move component and it's dependencies to compatibility module

### PR DESCRIPTION
**Pull request by elrodro83**
* test support, notifications

### CODE REVIEW

File: tests/unit/src/main/java/org/mule/tck/core/context/notification/AbstractNotificationLogger.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * LICENSE.txt file.
+ */
+package org.mule.tck.core.context.notification;
+
+import org.mule.runtime.core.api.context.notification.ServerNotification;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public abstract class AbstractNotificationLogger<T extends ServerNotification> implements NotificationLogger {
```
Comment (pablokraan) [inline tests/unit/src/main/java/org/mule/tck/core/context/notification/AbstractNotificationLogger.java]: Javadoc

File: tests/unit/src/main/java/org/mule/tck/core/context/notification/AbstractNotificationLogger.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  private LinkedList notifications = new LinkedList();
+
+  public synchronized void onNotification(T notification) {
```
Comment (pablokraan) [inline tests/unit/src/main/java/org/mule/tck/core/context/notification/AbstractNotificationLogger.java]: Missing override?

File: tests/unit/src/main/java/org/mule/tck/core/context/notification/NotificationLogger.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tck.core.context.notification;
+
+import java.util.List;
+
+public interface NotificationLogger {
```
Comment (pablokraan) [inline tests/unit/src/main/java/org/mule/tck/core/context/notification/NotificationLogger.java]: Add javadoc

File: tests/unit/src/main/java/org/mule/tck/junit4/AbstractMuleContextTestCase.java [hunk 4]
```diff
@@ -399,17 +396,16 @@ protected void doTearDown() throws Exception {
...
-    ((MuleContextAware) component).setMuleContext(muleContext);
-
-    Flow flow = builder(name, muleContext).processors(component).build();
+  public static Flow getTestFlow(String name) throws MuleException {
+    Flow flow = builder(name, muleContext).build();
     muleContext.getRegistry().registerFlowConstruct(flow);
     return flow;
   }
 
+  public static Flow getTestFlow(MuleContext muleContext) throws MuleException {
```
Comment (pablokraan) [inline tests/unit/src/main/java/org/mule/tck/junit4/AbstractMuleContextTestCase.java]: Is this method needed? Looks like we can just use MuleTestUtils's

File: tests/unit/src/main/java/org/mule/tck/core/context/notification/AbstractNotificationLogger.java (hunk 2 above)
Comment (elrodro83) [inline tests/unit/src/main/java/org/mule/tck/core/context/notification/AbstractNotificationLogger.java]: no, the implemented interface doesn't declare this method

**Pull request merged**